man/openmp-utils.Rd

```diff
@@ -38,20 +38,121 @@
 
   \itemize{
     \item\file{between.c} - \code{\link{between}()}
+    
+    OpenMP is used here to parallelize:
+    \itemize{
+      \item The loops that check if each element of the vector provided is between the specified \code{lower} and \code{upper} bounds, for integer (\code{INTSXP}) and real (\code{REALSXP}) types
+      \item The checking and handling of undefined values (such as NaNs)
+    }
+    
+    Since this function is used to find rows where a column's value falls within a specific range, it benefits more from parallelization when the input data consists of a large number of rows.
+    
     \item\file{cj.c} - \code{\link{CJ}()}
+    
+    OpenMP is used here to parallelize:
+
+    \itemize{
+      \item The element assignment in vectors
+      \item The memory copying operations (blockwise replication of data using \code{memcpy})
+      \item The creation of all combinations of the input vectors over the cross-product space
+    }
+    
+    Given that the number of combinations increases exponentially as more columns are added, better speedup can be expected when dealing with a large number of columns.
+    
     \item\file{coalesce.c} - \code{\link{fcoalesce}()}
+    
+    OpenMP is used here to parallelize:
+    \itemize{
+      \item The operation that iterates over the rows to coalesce the data (which can be of type integer, real, or complex)
+      \item The replacement of NAs with non-NA values from subsequent vectors
+      \item The conditional checks within parallelized loops
+      }
+    
+    Significant speedup can be expected for more number of columns here, given that this function operates efficiently across multiple columns to find non-NA values.
+    
     \item\file{fifelse.c} - \code{\link{fifelse}()}
+    
+    For logical, integer, and real types, OpenMP is being used here to parallelize loops that perform conditional checks along with assignment operations over the elements of the supplied logical vector based on the condition (\code{test}) and values provided for the remaining arguments (\code{yes}, \code{no}, and \code{na}).
+    
+    Better speedup can be expected for more number of columns here as well, given that this function operates column-wise with independent vector operations.
```

Comment:
I don't understand what is meant by more number of columns

Reply:
I meant that relative to columns (or that the speedup is better for more number of columns as opposed to having more number of rows in the input data) but you're right that it isn't obvious without the mention of the rows vs. columns thing. I also don't like how I'm repeating myself on that note for every function..do you think it would better to have a statement towards the end of the document enlisting all the functions that perform better with more number of rows vs the ones that perform better with more number of columns?

Reply:
I meant that relative to columns (or that the speedup is better for more number of columns as opposed to having more number of rows in the input data) but you're right that it isn't obvious without the mention of the rows vs. columns thing. I also don't like how I'm repeating myself on that note for every function..do you think it would be more preferable to have a statement towards the end of the document enlisting all the functions that perform better with more number of rows vs the ones that perform better with more number of columns?